js/core/src/action.ts

```diff
@@ -120,6 +122,10 @@ export function action<
       ? config.name
       : `${config.name.pluginId}/${config.name.actionId}`;
   const actionFn = async (input: I) => {
+    if (input?.hasOwnProperty(GENKIT_SESSION_STATE_INPUT_KEY)) {
+      input = { ...input };
```

Comment:
Hmm... Why doesn't this break: https://github.com/firebase/genkit/blob/64cd023d7c645857fe21c4c5404d0e58a1e27f1e/js/genkit/tests/session_test.ts#L342

I think you're only supposed to strip it for parsing, but add it back in for action fn call on line 144......